node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/nodeagent/NodeAgentImpl.java

```diff
@@ -193,15 +197,11 @@ public void stop() {
             }
         } while (loopThread.isAlive() || !filebeatRestarter.isTerminated());
 
+        healthChecker.ifPresent(HealthChecker::close);
```

Comment:
Its not clear from this code, but the `HealthChecker` is shared, so cannot be closed here.

Reply:
agreed